plugin/src/main/java/org/opensearch/ml/action/prediction/TransportPredictionTaskAction.java

```diff
@@ -227,4 +237,24 @@ private void executePredict(
                 })
             );
     }
+
+    private void validateInputSchema(String modelId, MLInput mlInput) {
+        if (modelCacheHelper.getModelInterface(modelId) != null && modelCacheHelper.getModelInterface(modelId).get("input") != null) {
+            String inputSchemaString = modelCacheHelper.getModelInterface(modelId).get("input");
+            try {
+                String parametersString = parametersObjectExtractor(mlInput);
+                MLNodeUtils.validateSchema(inputSchemaString, parametersString);
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Error validating input schema: " + e.getMessage());
+            }
+        }
+    }
+
+    private static String parametersObjectExtractor(MLInput mlInput) throws IOException {
+        String mlInputString = mlInput.toXContent(XContentFactory.jsonBuilder(), ToXContent.EMPTY_PARAMS).toString();
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode mlInputObject = mapper.readTree(mlInputString);
+        return mlInputObject.get("parameters").asText();
```

Comment:
By hard code `parameters` here, this function can only be used by remote model, and we can't extend for future use case, for example what if we add another field in model `input` like
```
{
    "parameters": {
        "input1": "..."
    },
    "new_field1": "..." // new field added 
}
```

Reply:
Removed